eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -193,6 +197,8 @@ jobs:
 
         steps:
           - template: ../steps/common.yml
+            parameters:
+              OSVmImage: $(OSVmImage)
```

Comment:
What prompted the switch to template parameters here?

Reply:
It's a template calling a template, so am just passing the parameter down to the next template. That isn't a template parameter, is it? Isn't that just a regular parameter? I thought template parameter are like ${{ parameters.name }}

Reply:
`$(OSVmImage)` is a normal variable but you are passing it as a template parameter where as before we just accessed it via the normal variable. What you have will work but I wasn't sure if there was something that made you introduce the template parameter now.

Reply:
I just needed to get it into common.yml since am referencing  the resource as a template.

`  - template: eng/pipelines/templates/scripts/verify-agent-os.yml@azure-sdk-tools
    parameters:
      OSName: ${{parameters.OSVmImage}}
`

Reply:
What you have will work but I believe you could of also just did:
```
- template: eng/pipelines/templates/scripts/verify-agent-os.yml@azure-sdk-tools 
  parameters: 
    OSName: $(OSVmImage)
```